File: Sliding_window/Java/Kadane's Algorithm.java
// Problem link: https://www.geeksforgeeks.org/problems/kadanes-algorithm-1587115620/1
// Time: O(n)
// Space: O(1)
class Solution{
    long maxSubarraySum(int arr[], int n){
        long maxSum = arr[0], currSum = 0;
        
        for(int i : arr){
            if(currSum < 0){
                currSum = 0;
            }
            currSum += i;
            maxSum = Math.max(maxSum, currSum);
        }
        
        return maxSum;
        
    }
    
}

File: Sliding_window/Python/Kadane's Algorithm.py
# Problem link: https://www.geeksforgeeks.org/problems/kadanes-algorithm-1587115620/1
# Time: O(n)
# Space: O(1)
class Solution:
    def maxSubArraySum(self,arr,N):
        maxSum, currSum = arr[0], 0
        
        for i in arr:
            if currSum < 0:
                currSum = 0
            currSum += i
            maxSum = max(maxSum, currSum)
        
        return maxSum
